std/portable/index.js

```diff
@@ -198,11 +198,39 @@ globalScope["changetype"] = function changetype(value) {
 };
 
 String["fromCharCodes"] = function fromCharCodes(arr) {
-  return String.fromCharCode.apply(String, arr);
+  const CHUNKSIZE = 1 << 15;
+  const len = arr.length;
+  if (len <= CHUNKSIZE) {
+    return String.fromCharCode.apply(String, arr);
+  }
+  let index = 0;
+  let parts = '';
+  while (index < len) {
+    parts += String.fromCharCode.apply(
+      String,
+      arr.slice(index, Math.min(index + CHUNKSIZE, len))
+    );
+    index += CHUNKSIZE;
+  }
+  return parts;
 };
 
 String["fromCodePoints"] = function fromCodePoints(arr) {
-  return String.fromCodePoint.apply(String, arr);
+  const CHUNKSIZE = 1 << 15;
```

Comment:
Also, my worries about making this too large aren't so much about speed, but about reasonably guaranteeing not to overrun the stack in a fix that is supposed to fix overrunning the stack. We can't really tell if these functions will be used somewhere deeper down the stack, even though it might work with `1 << 15` with how the compiler calls them.

Reply:
How about `1 << 14`? I guess it's quite optimal and allow up to 3 level stack nesting

Reply:
We can try if you wish, but chances are that people will run into problems and give up without ever reporting the problem. Perhaps I'm favoring being safe too much there?

Reply:
Alright) I guess 1 << 13 shouldn't cause any problems